Plugin.php

```diff
@@ -52,43 +60,167 @@ public function boot()
+    {
+        /** @var \Barryvdh\Debugbar\LaravelDebugbar $debugBar */
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        if (Config::get('debugbar.collectors.cms', true)) {
+            // Disable route collector as the CMS collector presents this info instead
+            Config::set('debugbar.collectors.route', false);
+
+            Event::listen('cms.page.beforeDisplay', function (CmsController $controller, $url, ?Page $page) use ($debugBar) {
+                if ($page) {
+                    $collector = new CmsCollector($controller, $url, $page);
+                    if (!$debugBar->hasCollector($collector->getName())) {
+                        $debugBar->addCollector($collector);
+                    }
+                }
+            });
+        }
+
+        if (Config::get('debugbar.collectors.components', true)) {
+            Event::listen('cms.page.initComponents', function (CmsController $controller, ?Page $page, ?Layout $layout) use ($debugBar) {
+                if ($page) {
+                    $collector = new ComponentsCollector($controller, $page, $layout);
+                    if (!$debugBar->hasCollector($collector->getName())) {
+                        $debugBar->addCollector($collector);
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Add collectors used by the Backend only
+     */
+    public function addBackendCollectors()
+    {
+        /** @var \Barryvdh\Debugbar\LaravelDebugbar $debugBar */
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        if (Config::get('debugbar.collectors.backend', true)) {
+            Event::listen('backend.page.beforeDisplay', function (BackendController $controller, $action, array $params) use ($debugBar) {
+                $collector = new BackendCollector($controller, $action, $params);
+                if (!$debugBar->hasCollector($collector->getName())) {
+                    $debugBar->addCollector($collector);
+                }
+            });
+        }
+    }
+
+    /**
+     * Registers extensions in the CMS Twig environment
+     */
+    protected function registerCmsTwigExtensions()
+    {
+        $profile = new Profile;
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        Event::listen('cms.page.beforeDisplay', function ($controller, $url, $page) use ($profile, $debugBar) {
             $twig = $controller->getTwig();
             if (!$twig->hasExtension(\Barryvdh\Debugbar\Twig\Extension\Debug::class)) {
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Debug($this->app));
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Stopwatch($this->app));
             }
+
+            if (!$twig->hasExtension(ProfilerExtension::class)) {
+                $twig->addExtension(new ProfilerExtension($profile));
+            }
```

Comment:
What causes that extension to be present or not?

Reply:
Not sure, but there is the possibility it is already included in another fashion (perhaps a plugin?), so this is probably the right call to check if it's already applied before applying it here. It is required for the Twig tab in the debug bar to work correctly.